sdk/include/reactos/shellutils.h

```diff
@@ -456,7 +456,8 @@ template<class B, class R> static HRESULT SHILCombine(B base, PCUIDLIST_RELATIVE
     return r ? S_OK : E_OUTOFMEMORY;
 }
 
-template<class T> static inline bool StrIsNullOrEmpty(T str) { return !str || !*str; }
+static inline bool StrIsNullOrEmpty(LPCSTR str) { return !str || !*str; }
+static inline bool StrIsNullOrEmpty(LPCWSTR str) { return !str || !*str; }
```

Comment:
Out of curiosity, why did you prefer overloads here?

Reply:
To call the best operator if someone passes in a `CComHeapPtr<WCHAR>`.

Reply:
With `(T str)`, it might make a copy? With `(T*str)`, not sure if it would work? With `(T&str)`, it would probably work.